python/tk_multi_workfiles/actions/new_file_action.py

```diff
@@ -79,8 +79,8 @@ def execute(self, parent_ui):
                 raise TankError(
                     "Unable to resolve template fields after folder creation!  This could mean "
                     "there is a mismatch between your folder schema and templates. Please "
-                    "contact us via {} if you need help fixing this.".format(
-                        support_url
+                    "contact us via {url} if you need help fixing this.\n\n{error}".format(
+                        url=support_url, error=str(e)
                     )
                 )
 
```

Comment:
Rather than building a string, raising it, catching it, parsing it, and then displaying it, I'd like to see this as a try/except/else block of logic for this specific error case, and then the previous logic with the `QtGui.QMessageBox.information` for the general exception handling farther down.

So, up here, you'll have something like:

```python
try:
    FileAction.create_folders...
except TankError as e:
    self._app.log_exception...
    msg_box = QtGui.QMessageBox( ... )
    msg_box.setDefaultButton...
    msg_box.exec_()
    return False
else:
    ...
```

You'll just skip this raise entirely by handling the exception you got here and returning. Then you don't need the logic below that's trying to dissect the exception text and present it nicely.

Reply:
Agreed, this way is much simpler -- fixed.

Reply:
Thanks! I'll look it over again this afternoon.